examples/cluster/experimental_cloud.py

```diff
@@ -0,0 +1,36 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""
+This is a very basic sample script for running things remotely.
+It requires `aws_credentials` file to be present next to it.
+
+On credentials file format see https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html#cli-configure-files-where
+"""
+
+import os
+import logging
+logging.basicConfig(format='%(asctime)s %(message)s')
+logger = logging.getLogger()
+logger.setLevel(logging.DEBUG)
+
+from modin.experimental.cloud import Provider, Cluster, get_connection
+
+
+aws = Provider(Provider.AWS, os.path.join(os.path.abspath(os.path.dirname(__file__)), 'aws_credentials'), 'us-west-1')
+with Cluster(aws, cluster_name='rayscale-test') as c:
```

Comment:
Is this the API we are using? I'm not sure I like it.

I'm much more in favor of having the user explicitly create a cluster at the top and reuse variables below.

```python
# Factory preferred for explicitness. Instead of __init__ they are 
# typing create, which means we are standing up a cluster for them.
development_cluster = cluster.create(aws, cluster_name='rayscale-test')
type(development_cluster)  # Cluster

# Use and reuse created Cluster objects.
with development_cluster.deploy():
    pass
```

There would be no implementation change, just API change. What are your thoughts?

Reply:
I'm not sure what exact API change you're proposing here... I can rename `def Cluster(...)` to `def create_cluster(...)` if that is clearer.

Also there should not be a real need for a user to call `.deploy()` (or `.spawn()` in current API) - spawning happens automatically in the background when you create a cluster, because that's a lengthy process.

Reply:
I agree about the comment for `deploy()`. What about something like this:

```python
from modin.experimental.cloud import cluster

development_cluster = cluster.create(aws, cluster_name='rayscale-test')

with development_cluster:
    pass
```

If at all possible we should avoid showing the user this:

```python
with cluster.create(aws, cluster_name='rayscale-test') as development_cluster:
    pass

...

with developmen_cluster:
    pass
```

It is inconsistent and we should strive for consistency. They are functionally identical, but it is clearer to use assignment than `as` and reusing that variable later.

Reply:
We cannot forbid users from doing the second way... but we can always use the first way in examples and docs. I've rewritten the example to follow it.